### DISCUSSION TITLE
Matrix strategy labels

### DISCUSSION THREAD

**Discussion by robjmills** [community/community#26221] [Actions]
<p>The matrix strategy is a great way to run different commands with the same starting steps, to DRY up a workflow. As far as I can tell though there’s no way to effectively label each run, unless i’m missing something?</p>
<p>Take this example:</p>
<pre><code>name: PHP lints and tests workflow
on:
  push:

jobs:
  php-lints-and-tests:
    name: PHPStan
    runs-on: ubuntu-latest
    strategy:
      matrix:
        args:
          - 'php -d memory_limit=-1 ./vendor/bin/phpstan analyse --no-progress --level=7 app/src
          - 'php -d display_errors=On -d memory_limit=512M vendor/bin/phpcs -p --encoding=utf-8 --standard=tests/phpcs-ruleset.xml --error-severity=1 app/src'
          - 'php -d display_errors=On -d memory_limit=512M -d date.timezone=UTC vendor/bin/phpunit -c phpunitall.xml.dist'
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Composer installp
        run: composer.phar install --prefer-dist

      - name: PHP
        run: ${{ matrix.args }}
</code></pre>
<p>This works fine,  but each job in the matrix is listed as the command. Yaml does support key value pairs but I can’t see how to use a key as the job label. Is this possible?</p>

**Comment by chrispat** [comment, marked as answer]
<p>You could do something like:</p>
<pre><code>name: PHP lints and tests workflow
on:
  push:

jobs:
  php-lints-and-tests:
    name: PHPStan ${{ matrix.legs.name }}
    runs-on: ubuntu-latest
    strategy:
      matrix:
        legs:
          - {name: 'meaningful name', args: 'php -d memory_limit=-1 ./vendor/bin/phpstan analyse --no-progress --level=7 app/src' }
          - {name: 'meaningful name 2', args: 'php -d display_errors=On -d memory_limit=512M vendor/bin/phpcs -p --encoding=utf-8 --standard=tests/phpcs-ruleset.xml --error-severity=1 app/src' }
          - {name: 'meaningful naem 3', args: 'php -d display_errors=On -d memory_limit=512M -d date.timezone=UTC vendor/bin/phpunit -c phpunitall.xml.dist' }
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Composer installp
        run: composer.phar install --prefer-dist

      - name: PHP
        run: ${{ matrix.legs.args }}
</code></pre>
<p>legs is arbitrary could be any key name you like as is name.  But hopefully you get the idea and it helps you.</p>

**Comment by robjmills** [comment]
<p>Hi,</p>
<p>thanks for that but in the UI the Job label is still displayed like this:</p>
<p><img src="https://user-images.githubusercontent.com/110251/181096128-587c2dfc-b4a1-446b-a5c0-6d51ec1517cc.png" alt="stan.png" data-base62-sha1="xNL35a5OP9viOkYBNGaEQks8EqC" width="468" height="324"></p>
<p>It doesn’t look like the legs.name property is used anywhere.</p>

**Comment by Yanjingzhu** [comment]
<p>Please check your workflow yml, is there any typos?</p>
<p>I tested chrispat’s yml, the job name works fine (please ignore the failure in PHP step). </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2baa4fdf2b2375699cf08475da05107870e41acb.png" data-download-href="/uploads/short-url/6ehqe1MIddNcnvZmdJkwP5v3Jph.png?dl=1" title="matrix yml.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181096131-be535bce-6ad9-4245-85a0-98d523037858.png" alt="matrix yml.png" data-base62-sha1="6ehqe1MIddNcnvZmdJkwP5v3Jph" width="690" height="349" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2baa4fdf2b2375699cf08475da05107870e41acb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">matrix yml.png</span><span class="informations">877×444 10.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4728e6aeadb9dc8b75c0616728e99d8b6369cee9.png" data-download-href="/uploads/short-url/a9vwTFnu3a3oURWZMBsHz4g2akV.png?dl=1" title="matrix name.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181096132-a9a50fd9-d091-4a3f-894d-30c040e84aeb.png" alt="matrix name.png" data-base62-sha1="a9vwTFnu3a3oURWZMBsHz4g2akV" width="690" height="439" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4728e6aeadb9dc8b75c0616728e99d8b6369cee9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">matrix name.png</span><span class="informations">1035×659 23.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by robjmills** [comment]
<p>It did work after all. I had 2 versions of the same workflow and I was checking the wrong one. Oh dear!!</p>
